### PR TITLE
Fix check selection

### PR DESCRIPTION
**Pull request by willynganga**

### CODE REVIEW

File: src/test/java/com/acrolinx/sidebar/pojo/settings/CheckOptionsTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        CheckOptions checkOptions = getCheckOptions();
+
+        Assertions.assertEquals(
+                "{\"inputFormat\":\"AUTO\",\"requestDescription\":{\"documentReference\":\"foo\"},\"selection\":{\"ranges\":[[0,0]]},\"externalContent\":{}}",
+                checkOptions.toString());
+    }
+
+    @Test
+    void testToStringFailsIfNoValidKeysProvided()
```
Comment (Thilo-Acrolinx) [inline src/test/java/com/acrolinx/sidebar/pojo/settings/CheckOptionsTest.java:23]: I don't understand the purpose of this test method. We create a CheckOptions instances via the same helper method as in the first test method, so what scenario is this one covering that the first one does not already cover?

File: src/test/java/com/acrolinx/sidebar/pojo/document/CheckContentTest.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/* Copyright (c) 2018-present Acrolinx GmbH */
+package com.acrolinx.sidebar.pojo.document;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (Thilo-Acrolinx) [inline src/test/java/com/acrolinx/sidebar/pojo/document/CheckContentTest.java:4]: please configure your IDE to not use Wildcard imports (on all projects)

(same in a few other classes below)

File: src/test/java/com/acrolinx/sidebar/pojo/document/CheckedDocumentPartTest.java [hunk 3]
```diff
@@ -28,4 +29,13 @@ void getAsJS()
                 + "\"type\":\"ditaReferences\",\"originalBegin\":25,\"originalEnd\":35,\"externalContentMatches\":[{\"id\":\"chapter11.dita\",\"type\":\"ditaReferences\",\"originalBegin\":75,\"originalEnd\":90,\"externalContentMatches\":[]}]}]}",
                 partWithExternalContent.getAsJS());
     }
+
+    @Test
+    void toStringTest()
```
Comment (Thilo-Acrolinx) [inline src/test/java/com/acrolinx/sidebar/pojo/document/CheckedDocumentPartTest.java:34]: can you please rename the above method to "getAsJSTest" for consistency?

File: src/test/java/com/acrolinx/sidebar/pojo/document/CheckContentTest.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class CheckContentTest
+{
+
```
Comment (Thilo-Acrolinx) [inline src/test/java/com/acrolinx/sidebar/pojo/document/CheckContentTest.java:11]: nit-picking: unnecessary empty line, also in a few other classes below

you can use the following regular expression to quickly find them all:

`\{(\s*\n){2,}`

File: src/main/java/com/acrolinx/sidebar/pojo/settings/CheckOptions.java [hunk 5]
```diff
@@ -12,20 +12,20 @@ public class CheckOptions
 {
     private final InputFormat inputFormat;
     private final RequestDescription requestDescription;
-    private final DocumentSelection documentSelection;
+    private final DocumentSelection selection;
```
Comment (Thilo-Acrolinx) [inline src/main/java/com/acrolinx/sidebar/pojo/settings/CheckOptions.java:15]: another way to fix this problem would be to add a

`@SerializedName("selection")`

annotation to this field, then we could name it whatever we want (e.g. "documentSelection") but Gson would serialize it as "selection" in the JSON string.

Not sure which solution I prefer, what do you think?

Comment (willynganga) [reply src/main/java/com/acrolinx/sidebar/pojo/settings/CheckOptions.java:15]: Both work for me. I have no issue using `@SerializedName("selection")`.

File: src/test/java/com/acrolinx/sidebar/pojo/settings/AcrolinxSidebarInitParameterTest.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+class AcrolinxSidebarInitParameterTest
+{
+    @Test
+    void testToString()
+    {
+        String osName = System.getProperty("os.name", "Win");
+        String osVersion = System.getProperty("os.version", "13.5.1");
+        String javaVendor = System.getProperty("java.vendor", "Zuul");
+        String javaVmName = System.getProperty("java.vm.name", "OpenJDK Runtime Environment");
```
[the remark this answers is not in the document]
Comment (willynganga) [reply src/test/java/com/acrolinx/sidebar/pojo/settings/AcrolinxSidebarInitParameterTest.java:16]: Line 67 of AcrolinxSidebarInitParameter

[the remark this answers is not in the document]
Comment (Thilo-Acrolinx) [reply src/test/java/com/acrolinx/sidebar/pojo/settings/AcrolinxSidebarInitParameterTest.java:16]: Right, I did not see this first, thanks.

File: src/test/java/com/acrolinx/sidebar/pojo/settings/AcrolinxSidebarInitParameterTest.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2018-present Acrolinx GmbH */
+package com.acrolinx.sidebar.pojo.settings;
+
+import java.util.Collections;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class AcrolinxSidebarInitParameterTest
+{
+    public static final String OS_NAME = "os.name";
```
Comment (Thilo-Acrolinx) [inline src/test/java/com/acrolinx/sidebar/pojo/settings/AcrolinxSidebarInitParameterTest.java:10]: all constants could be private I think

File: src/test/java/com/acrolinx/sidebar/pojo/settings/SystemProperties.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2018-present Acrolinx GmbH */
+package com.acrolinx.sidebar.pojo.settings;
+
+public class SystemProperties
+{
+    String osName;
```
Comment (Thilo-Acrolinx) [inline src/test/java/com/acrolinx/sidebar/pojo/settings/SystemProperties.java:6]: all fields should be private

File: src/test/java/com/acrolinx/sidebar/pojo/settings/AcrolinxSidebarInitParameterTest.java [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    public static final String OS_VERSION = "os.version";
+    public static final String JAVA_VENDOR = "java.vendor";
+    public static final String JAVA_VM_NAME = "java.vm.name";
+    public static final String JAVA_RUNTIME_NAME = "java.runtime.name";
+    public static final String JAVA_RUNTIME_VERSION = "java.runtime.version";
+
+    @Test
+    void toStringTest()
+    {
+        SystemProperties systemProperties = createSystemProperties();
```
Comment (Thilo-Acrolinx) [inline src/test/java/com/acrolinx/sidebar/pojo/settings/AcrolinxSidebarInitParameterTest.java:20]: I am not quite happy with this design. The SystemProperties is mutable, which is ugly. We could add a SystemPropertiesBuilder class and make the SystemProperties immutable, but this adds even more boilerplate code.

Maybe a simpler solution could be to omit this SystemProperties class altogether and instead call `java.lang.System.getProperties()` which returns an `java.util.Properties` instance. We then could make a copy/clone of this Properties to store all the properties before we make changes. Then in the try-block we call `java.lang.System.setProperties(Properties)` to restore the original ones. What do you think?

Comment (willynganga) [reply src/test/java/com/acrolinx/sidebar/pojo/settings/AcrolinxSidebarInitParameterTest.java:20]: Isn't this what you had implied? Did I get it wrong?

File: src/test/java/com/acrolinx/sidebar/pojo/settings/AcrolinxSidebarInitParameterTest.java [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    private static final String OS_VERSION = "os.version";
+    private static final String JAVA_VENDOR = "java.vendor";
+    private static final String JAVA_VM_NAME = "java.vm.name";
+    private static final String JAVA_RUNTIME_NAME = "java.runtime.name";
+    private static final String JAVA_RUNTIME_VERSION = "java.runtime.version";
+
+    @Test
+    void toStringTest()
+    {
+        Properties properties = (Properties) System.getProperties().clone();
```
Comment (Thilo-Acrolinx) [inline src/test/java/com/acrolinx/sidebar/pojo/settings/AcrolinxSidebarInitParameterTest.java:21]: maybe make this local variable final to make the idea more clear and avoid accidental re-assigning further down

**Review by Thilo-Acrolinx — APPROVED** [review]

**Pull request merged**